example/lib/bloc/airship_bloc.dart

```diff
@@ -0,0 +1,154 @@
+  StreamSink<List<InboxMessage>> get messagesSink => _messagesStreamController.sink;
+
+  StreamSink<List<String>> get tagsAddedSink => _tagsAddedStreamController.sink;
+  StreamSink<List<String>> get tagsRemovedSink => _tagsRemovedStreamController.sink;
+
+  StreamSink<InboxMessage> get messageRemovedSink => _messageRemovedStreamController.sink;
+
+  AirshipBloc() {
+    Airship.channelId.then((val) {
+      _channelStreamController.add(val);
+    });
+
+    Airship.tags.then((val) {
+      _tagsStreamController.add(val);
+    });
+
+    _tagsAddedStreamController.stream.listen((tags) {
+      Airship.addTags(tags).then((val) {
+        // Add the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    _tagsRemovedStreamController.stream.listen((tags) {
+      Airship.removeTags(tags).then((val) {
+        // Remove the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    Airship.namedUser.then((val) {
+      _namedUserStreamController.add(val);
+    });
+
+    _namedUserSetStreamController.stream.listen((namedUser) {
+      Airship.setNamedUser(namedUser).then((val) {
+        namedUserSink.add(namedUser);
+      });
+    });
+
+    Airship.userNotificationsEnabled.then((val) {
+      _notificationsEnabledStreamController.add(val);
+    });
+
+    _notificationsEnabledSetStreamController.stream.listen((enabled) {
+      Airship.setUserNotificationsEnabled(enabled).then((val) {
+        notificationsEnabledSink.add(enabled);
+      });
+    });
+
+    Airship.inboxMessages.then((messages) {
+      _messagesStreamController.add(messages);
+    });
+
+    _messageAddedStreamController.stream.listen((messages) {
+      Airship.inboxMessages.then((messages) {
+        messagesSink.add(messages);
+      });
+    });
+
+    _messageRemovedStreamController.stream.listen((message) {
+      Airship.deleteInboxMessage(message).then((onValue) {
+        // Get fresh message list
+        Airship.inboxMessages.then((messages) {
+          messagesSink.add(messages);
+        });
+      });
+    });
+
+    Airship.onInboxUpdated.listen((event) {
+      debugPrint('Inbox updated link');
+      // Get fresh message list
+      Airship.inboxMessages.then((messages) {
+        messagesSink.add(messages);
+      });
+    });
+
+    Airship.onShowInboxMessage.listen((messageId) {
+      //MaterialPageRoute(builder: (context) => MessageView(message: message));
+      debugPrint('Show inbox message $messageId');
+      Airship.inboxMessages.then((messages) {
+        messagesSink.add(messages);
+      });
+    });
+
+    Airship.onChannelRegistration.listen((event) {
+      _channelUpdatedStreamController.add(event.channelId);
+      debugPrint('Channel registration $event');
+    });
+
+    Airship.onPushReceived.listen((event) {
+      debugPrint('Push Received $event');
```

Comment:
Should there be a stream for `onPushReceived`? Does the app just listen to `Airship.onPushReceived` elsewhere? These questions apply to the next three events also.

Reply:
Going to flesh these out in this follow up ticket:
https://urbanairship.atlassian.net/browse/MOBILE-849

I think this greatly improves on what we have. Adding deep linking is going to add some complexity and I think it's best to tackle it in a separate PR so it doesn't become impossible to review.